cpp/src/barretenberg/proof_system/plookup_tables/dummy.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+/**
+ * @file dummy.hpp
+ * @author Rumata888
+ * @brief This file contains functions for the dummy tables that we use in UltraHonk to make table, sorted and lookup
+ * selector polynomials non-zero.
+ *
+ */
+
+#include "types.hpp"
+
+namespace plookup {
+namespace dummy_tables {
+
+/**
+ * @brief Lookup the value corresponding to a sepcific key
+ *
+ * @details We need this function for when we are constructing the circuit and want to query the table. Since we need
+ * two basic tables to make the table polynomial have non-zero values, we instantiate two tables with the same function,
+ * but change it slightly through templating
+ *
+ * @tparam id The id of the basic table used to parametrize the values for 2 fake tables
+ * @param key The key that we are looking up
+ * @return std::array<barretenberg::fr, 2>
+ */
+template <uint64_t id> inline std::array<barretenberg::fr, 2> get_value_from_key(const std::array<uint64_t, 2> key)
+{
+    return { key[0] * 3 + key[1] * 4 + id * 0x1337ULL, 0ULL };
+}
+
+/**
+ * @brief Generate the whole table
+ *
+ * @details This function is used to generate the whole table for the table polynomial. It's templated with id, since we
+ * generate 2 slightly different fake tables.
+ *
+ * @tparam table_id The id of the table this function is instantiated for
+ * @param id Table id that is the same for all circuits
+ * @param table_index The index for this table that is used in this circuit. 0, 1, ...
+ * @return A table of values
+ */
+template <uint64_t table_id>
+inline BasicTable generate_honk_dummy_table(const BasicTableId id, const size_t table_index)
+{
+
+    ASSERT(table_id == static_cast<uint64_t>(id));
```

Comment:
why do this two have to be equal - i understand that `id` is static while `table_id` can change depending on function (?)

Reply:
We are using the same function in 2 places by templating it, but the API already includes the id in arguments. This is just a simple bug check to ensure that we don't use the wrong instantiation